proposals/0000-extract-runtime.md

```diff
@@ -0,0 +1,419 @@
+with the rest of the React Native and the distinction can be a bit confusing for newcomers.
+
+Such blocks would have access to Event Loop and JS engine, so new globals can be added in as-needed
+basis.
+
+Taking WebAPIs as an example, having their implementations split across multiple different TurboModules,
+we can release and version them independently further simplifying update process.
+In case the `react-natime-runtime` changes in a breaking manner, we can tightly couple the versions
+of such TurboModules or make them part of `react-native-runtime` that are extracted later, when
+the breaking changes become less frequent.
+
+__Secondary goal is to be able to develop and test those headless modules easily, on the CI pipeline without visual environment__.
+
+By having the possibility to run the "blocks" (C++ only turbo modules) in a headless JS environment we
+are able to test the implementations easily on many CI platforms and on different JS engines (ideally).
+
+A side effect of this is we can perhaps achieve faster build times both on CI and locally, resulting in lower cost of such pipelines.
+
+__Tertiary goal is to bring OOT (out of tree) platforms closer to first-class citizens__
+
+If a Turbo Module does not need access to UI in any way to provide its features, having it work with `react-native-runtime`
+rather than `react-native` would make it easier to maintain for currently OOT platforms (such as rn-windows and rn-mac) where
+the current `react-native` code cannot be re-used in 100% (needs further clarification). 
+Right now we are biased towards mobile platforms only.
+
+### Non-goals
+
+This RFC does **not** aim to:
+
+ - Replace or deprecate proposal [Move iOS and Android Specific Code to Their Own Packages](https://github.com/react-native-community/discussions-and-proposals/pull/49)
+
+## Detailed design
+
+> **NOTE**: Because we are talking about headless code, TurboModules refer to C++ only TurboModules unless
+> otherwise noted.
+
+### Overview
+
+Before proceeding to the description of the suggested changes, let's consider an example of a React Native application 
+with additional OOT platform can be represented in current state (more or less):
+
+![Current State Diagram](./assets/0000-rn-diagram-current-oot.png)
+
+The proposed `react-native-runtime` layer would be everything from React Native that is
+independent of the UI layer (so essentially everything headless) with enough extension points and mount
+points to make it elastic enough to be extended by UI layer or platform layer, if needed.
+
+This includes, but is not limited to:
+
+ - Event Loop
+ - JS engine management
+ - C++-only Turbo Module hosting
+
+And ideally, in the future (out of scope for this RFC):
+ - DevTools interface and debugging host (long term, after the interface gets stabilized)
+ - Handling of Shared and Dedicated Workers
+
+`react-native` would then focus mostly on:
+
+  - Shadow Nodes
+  - Rendering
+  - UI TurboModules (?)
```

Comment:
I guess you mean something like "Turbo Modules exposing platform-native functionality"

Reply:
Updated the wording